jenkins/shared-library/vars/defaultCiPipeline.groovy

```diff
@@ -67,6 +67,11 @@ void call(Closure body) {
 		}
 
 		stages {
+			stage('setup environment') {
+				steps {
+					runScript('git submodule update --remote')
+				}
+			}
```

Comment:
why do we need this here in addition to helper.groovy?

Reply:
The one in the helper is only for manual jobs.  This one will always run.
I was not sure if the ``git reset hard`` also reset the submodule.
From my testing it seems it does not but not sure.